src/index.ts

```diff
@@ -162,6 +173,15 @@ export default class Collection<Resource extends { [key: string]: any }> {
     }, col);
   }
 
+  public concat(idProp: string, other: Collection<Resource>): Collection<Resource> {
```

Comment:
Can we assert at a type level that `idProp` is a key in `Resource`?

Reply:
Oh, interesting. I *think* so!

Reply:
:zap: 4711ae1

```ts
interface Item {
  id: string;
  foo: string;
}

const col = new Collection<Item>().withList('id', items); // works!
const bad = new Collection<Item>().withList('di', items); // compiler error!
// Argument of type '"di"' is not assignable to parameter of type '"id" | "foo"'.
```